DigitalLearningSolutions.Web/Attributes/DelegateOnlyInaccessibleAttribute.cs

```diff
@@ -0,0 +1,21 @@
+﻿namespace DigitalLearningSolutions.Web.Attributes
+{
+    using System;
+    using DigitalLearningSolutions.Web.Helpers;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.AspNetCore.Mvc.Filters;
+
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method)]
+    public class DelegateOnlyInaccessibleAttribute : Attribute, IAuthorizationFilter

```

Comment:
We have custom policies for the Authorize attributes that redirect to the Log In page, which each policy requiring different claims to allow access to the page (e.g. UserCentreAdminOnly policy used on the TrackingSystem/CentreConfiguration controller). Is it possible to have this written as a custom policy so we have all that logic together? I suspect the policies do not allow specific redirects based on the policy so it might not be feasible - if that's the case then this is fine.

Reply:
That the custom policy doesn't allow a different redirect is the conclusion Stella and I came to when we looked at this possibility before, but I'll take a specific look.

Reply:
I've not found any indication this can be done simply with a policy.